Improve symlink dirs documentation (#369)
cylc/cylc-doc: src/user-guide/installing-workflows.rst

``~/cylc-src/dogs/fido``, or, failing that, ``~/roses/dogs/fido``, and install
the first match (into ``~/cylc-run/dogs/fido/run1``).

.. _SymlinkDirs:

Symlink Directories
^^^^^^^^^^^^^^^^^^^

You can configure workflow :term:`run directories <run directory>` and certain
sub-directories as symlinks to other locations. This is a useful way of
offloading data onto other drives to limit the disk space taken up by
``~/cylc-run``.

Directories that can be individually symlinked are:

* ``log``
* ``share``
* ``share/cycle``
* ``work``
* the :term:`run directory` itself

> They are configured in :cylc:conf:`global.cylc[install][symlink dirs]`.

Reviewer: ```suggestion
The targets of symlink directories are set using :cylc:conf:`global.cylc[install][symlink dirs]`.
```

> Cylc will automatically create these symlinks on install;

Reviewer: seriously pedantic
```suggestion
Cylc will create these symlinks on workflow install;
```

Author: Actually decided to remove this line as it only applies for localhost symlink dirs

> see :ref:`Symlinking of directories`.

Reviewer: If you decide to keep the numbers https://github.com/cylc/cylc-doc/pull/369/files#r781123840 you might want to do this to prevent the number actually appearing here.
```suggestion
see :ref:`Symlinking of directories <Symlinking of directories>`.
```

> For example, to configure workflow ``log`` directories (on your local machine)

Reviewer 2: `scheduler` rather than on your local machine?

Example: ``global.cylc``:

.. code-block:: cylc

   [install]
       [[symlink dirs]]

>            [[[localhost]]]

Reviewer: Ignorant question: What if this isn't localhost? A look at the config suggests that data will be saved to another install target: Does this functionality exist? Will it in future? If not, why have this level of configuration?

Author: I have updated the lead-in to this example, does that explain it better?

> 2. Symlinking of Directories

Reviewer: I'm not totally sure that this and number 1 should be numbered, at least not explicitly.
